src/LanguageServer/Impl/Completion/CompletionItemSource.cs

```diff
@@ -47,19 +47,23 @@ public CompletionItem CreateCompletionItem(string text, IMember member, IPythonT
                 docFormat = InsertTextFormat.Snippet;
             }
 
-            return new CompletionItem {
+            return new CompletionItemEx {
                 label = label ?? text,
                 insertText = text,
                 insertTextFormat = docFormat,
                 // Place regular items first, advanced entries last
                 sortText = char.IsLetter(text, 0) ? "1" : "2",
                 kind = kind,
-                documentation = !t.IsUnknown() ? _docSource.GetHover(label ?? text, member, self) : null
+                documentation = !t.IsUnknown() ? _docSource.GetHover(label ?? text, member, self) : null,
+                // Custom fields used by the LS extensions that may modify
+                // the completion list. Not passed to the client.
+                Member = member,
```

Comment:
Does this mean that they're going to continue to be reliant on the internal details of the analyzer?

Reply:
These are public interfaces, i,e, `IMember`, `IPythonType`, `IDocumentAnalysis`

Reply:
I assume that if we modify those types, then they'd break? I had just thought that us splitting apart the analysis, the language server, and the parser into their own assemblies was to prevent breakage, but I guess if they need it then there's probably no recourse.

Reply:
We dont modify public interfaces after they are published. It would be `IDocumentAnalysis2` etc